From 35bb3c02126d76df5f76da5f2672f473e8e28f4f Mon Sep 17 00:00:00 2001
From: Vladimir Lelicanin <v.lelicanin@sae.edu>
Date: Sat, 5 May 2018 23:19:20 +0200
Subject: [PATCH 13/22] Fix/travis fails no wallet (#102)

 Fix Travis fails because there is no wallet to check
---
 ot-node.js | 44 +++++++++++++++++++++++---------------------
 1 file changed, 23 insertions(+), 21 deletions(-)

diff --git a/ot-node.js b/ot-node.js
index 2e3f319..d527ed6 100644
--- a/ot-node.js
+++ b/ot-node.js
@@ -85,28 +85,30 @@ class OTNode {
         }
 
         // check does node_wallet has sufficient Ether and ATRAC tokens
-        try {
-            const etherBalance = deasync(Utilities.getBalanceInEthers());
-            if (etherBalance <= 0) {
-                console.log('Please get some ETH in the node wallet before running ot-node');
-                process.exit(1);
-            } else {
-                (
-                    log.info(`Initial balance of ETH: ${etherBalance}`)
-                );
-            }
-
-            const atracBalance = deasync(Utilities.getAlphaTracTokenBalance());
-            if (atracBalance <= 0) {
-                console.log('Please get some ATRAC in the node wallet before running ot-node');
-                process.exit(1);
-            } else {
-                (
-                    log.info(`Initial balance of ATRAC: ${atracBalance}`)
-                );
+        if (process.env !== 'test') {
+            try {
+                const etherBalance = deasync(Utilities.getBalanceInEthers());
+                if (etherBalance <= 0) {
+                    console.log('Please get some ETH in the node wallet before running ot-node');
+                    process.exit(1);
+                } else {
+                    (
+                        log.info(`Initial balance of ETH: ${etherBalance}`)
+                    );
+                }
+
+                const atracBalance = deasync(Utilities.getAlphaTracTokenBalance());
+                if (atracBalance <= 0) {
+                    console.log('Please get some ATRAC in the node wallet before running ot-node');
+                    process.exit(1);
+                } else {
+                    (
+                        log.info(`Initial balance of ATRAC: ${atracBalance}`)
+                    );
+                }
+            } catch (error) {
+                console.log(error);
             }
-        } catch (error) {
-            console.log(error);
         }
 
         // wire instances
-- 
2.13.5 (Apple Git-94)

